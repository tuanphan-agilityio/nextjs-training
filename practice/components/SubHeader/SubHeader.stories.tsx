import { Story, Meta } from '@storybook/react';

import SubHeader, { SubHeaderProps } from '.';

export default {
  title: 'SubHeader',
  component: SubHeader,
} as Meta;

const Template: Story<SubHeaderProps> = (args) => <SubHeader {...args} />;

export const Default = Template.bind({});
Default.args = {
  breadcrumbItems: [
    { label: 'Home', href: '/' },
    { label: 'Category', href: '/category' },
    { label: 'Subcategory', href: '/category/subcategory' },
  ],
  title: 'Page Title',
  description: 'This is a subheader description.',
};
