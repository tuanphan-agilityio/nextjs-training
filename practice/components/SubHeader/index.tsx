import { FC, memo } from 'react';
import Breadcrumb, { BreadcrumbItem } from '@/components/Breadcrumb';

interface SubHeaderProps {
  breadcrumbItems: BreadcrumbItem[];
  title: string;
  description?: string;
}

const SubHeader: FC<SubHeaderProps> = ({
  breadcrumbItems,
  title,
  description = '',
}) => {
  return (
    <section className='pt-10 pb-20 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary'>
      <div className='container'>
        <Breadcrumb items={breadcrumbItems} />
        <h1 className='mt-10 text-3xl font-secondary-bold leading-4'>
          {title}
        </h1>
        {description && (
          <p className='mt-6 w-[508px] font-md leading-2'>{description}</p>
        )}
      </div>
    </section>
  );
};

export type { SubHeaderProps };

export default memo(SubHeader);
