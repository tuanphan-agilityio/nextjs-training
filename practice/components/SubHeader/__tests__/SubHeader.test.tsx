import { render } from '@testing-library/react';

import SubHeader, { SubHeaderProps } from '..';

describe('SubHeader Component', () => {
  const mockProps: SubHeaderProps = {
    breadcrumbItems: [
      { label: 'Home', href: '/' },
      { label: 'Products', href: '/products' },
    ],
    title: 'Product Page',
    description: 'Browse our products',
  };

  it('renders correctly with mockProps', () => {
    const { container } = render(<SubHeader {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('displays the breadcrumb', () => {
    const { getByText } = render(<SubHeader {...mockProps} />);
    mockProps.breadcrumbItems.forEach((item) => {
      const breadcrumbText = getByText(item.label);
      expect(breadcrumbText).toBeInTheDocument();
    });
  });

  it('does not display description when not provided', () => {
    const { queryByText } = render(
      <SubHeader
        breadcrumbItems={mockProps.breadcrumbItems}
        title={mockProps.title}
      />,
    );
    const descriptionText = queryByText(mockProps.description as string);
    expect(descriptionText).toBeNull();
  });
});
