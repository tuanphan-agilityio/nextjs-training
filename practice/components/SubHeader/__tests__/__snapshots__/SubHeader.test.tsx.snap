// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader Component renders correctly with mockProps 1`] = `
<div>
  <section
    class="pt-10 pb-20 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary"
  >
    <div
      class="container"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="flex w-fit text-sm text-primary"
        >
          <li
            class=""
          >
            <a
              class="hover:underline"
              href="/"
            >
              Home
            </a>
            <span
              class="mx-6"
            >
              /
            </span>
          </li>
          <li
            class="font-primary-bold"
          >
            Products
          </li>
        </ol>
      </nav>
      <h1
        class="mt-10 text-3xl font-secondary-bold leading-4"
      >
        Product Page
      </h1>
      <p
        class="mt-6 font-md leading-2"
      >
        Browse our products
      </p>
    </div>
  </section>
</div>
`;
